roman_kuprii/main.cpp

```diff
@@ -1,383 +1,57 @@
-        }
-        if (red_vertices_ids.size() > 0) {
-          std::uniform_int_distribution<> distr(0, red_vertices_ids.size() - 1);
-          work_graph.connect_vertices(start_vertex.id,
-                                      red_vertices_ids[distr(gen)], false);
-        }
-      }
-    }
-  }
-}
+int main() {
+  auto& logger = Logger::get_logger();
+  logger.set_output(LOG_FILENAME);
 
-void add_yellow_edges(Graph& work_graph) {
-  const int graph_depth = work_graph.get_depth();
-  std::random_device rd;
-  std::mt19937 gen(rd());
-  for (const auto& start_vertex : work_graph.get_vertices()) {
-    const double probability = static_cast<double>(start_vertex.depth) /
-                               static_cast<double>(graph_depth);
-    if (get_random_number() < probability) {
-      vector<VertexId> yellow_vertices_ids;
-      for (const auto& end_vertex : work_graph.get_vertices()) {
-        if (end_vertex.depth == start_vertex.depth + 1) {
-          if (!work_graph.is_connected(start_vertex.id, end_vertex.id))
-            yellow_vertices_ids.push_back(end_vertex.id);
-        }
-      }
-      if (yellow_vertices_ids.size() > 0) {
-        std::uniform_int_distribution<> distr(0,
-                                              yellow_vertices_ids.size() - 1);
-        work_graph.connect_vertices(start_vertex.id,
-                                    yellow_vertices_ids[distr(gen)], false);
-      }
-    }
-  }
-}
+  int graphs_quantity = GRAPHS_NUMBER;
 
-void paint_edges(Graph& work_graph) {
-  add_blue_edges(work_graph);
-  add_green_edges(work_graph);
-  add_red_edges(work_graph);
-  add_yellow_edges(work_graph);
-}
+  do {
+    std::cout << "Enter amount of graphs to generate" << std::endl;
+    std::cin >> graphs_quantity;
+  } while (graphs_quantity < GRAPHS_NUMBER);
 
-int main() {
   int depth = INVALID_NEW_DEPTH;
+
   do {
-    std::cout << "Enter generate graph depth from zero" << endl;
+    std::cout << "Enter generate graph depth from zero" << std::endl;
     std::cin >> depth;
   } while (depth <= INVALID_NEW_DEPTH);
+
   int new_vertices_num = INVALID_NEW_VERTICES_NUM;
+
   do {
-    std::cout << "Enter new_vertices_num from zero" << endl;
+    std::cout << "Enter new_vertices_num from zero" << std::endl;
     std::cin >> new_vertices_num;
   } while (new_vertices_num < INVALID_NEW_VERTICES_NUM);
 
-  std::cout << "Depth of adding vertices: " << depth << endl;
+  std::cout << "Depth of adding vertices: " << depth << std::endl;
 
-  Graph my_graph;
-  my_graph.add_vertex();
-  new_vertices_generation(my_graph, depth, new_vertices_num);
-  paint_edges(my_graph);
-  write_graph(my_graph);
+  for (int graph_num = 0; graph_num < graphs_quantity; graph_num++) {
+    Graph my_graph;
+    my_graph.add_vertex();
+    graph_generating::new_vertices_generation(my_graph, depth,
+                                              new_vertices_num);
+    graph_generating::paint_edges(my_graph);
```

Comment:
Какой смысл в генераторе, если ты не пользуешься удобным интерфейсом, а можно сказать, пишешь процедурную логику по вызову последовательности функций. Где же здесь удобство?
```suggestion
    const auto graph = uni_cpp_practice::graph_generation::generate(depth, new_vertices_num);
```